manifests/service.pp

```diff
@@ -6,15 +6,16 @@
 class promtail::service {
   case $facts['kernel'] {
     'Linux': {
-      systemd::unit_file { 'promtail.service':
-        source => 'puppet:///modules/promtail/promtail.service',
-        notify => Service['promtail'],
+      if $promtail::install_method == 'archive' {
+        systemd::unit_file { 'promtail.service':
+          source => 'puppet:///modules/promtail/promtail.service',
+          notify => Service['promtail'],
+        }
       }
 
       service { 'promtail':
-        ensure  => $promtail::service_ensure,
-        enable  => $promtail::service_enable,
-        require => Systemd::Unit_file['promtail.service'],
```

Comment:
Why is this line being removed?

Reply:
If you use a package the systemd unit file would be managed by the package and not as a resource in Puppet.

Reply:
In that case, we should use a selector or similar to set the contents of the require to the old value when installed not by package and to undef when installed by a package.

Reply:
It seems to me that it would make sense then to add a `before` param to the unit file then...